src/pages/Account/Tabs/ModulesTab.tsx

```diff
@@ -55,19 +83,169 @@ function ModulesContent({
   );
 }
 
-type ModulesTabProps = {
+function ModulesContentReworked({address}: {address: string}): JSX.Element {
+  const {data: registry} = useGetAccountResource(
+    address,
+    "0x1::code::PackageRegistry",
+  );
+
+  const [selectedModuleIndex, setSelectedModuleIndex] =
+    React.useState<number>(0);
+
+  const packages: PackageMetadata[] =
+    registry === undefined ? [] : (registry.data as any).packages;
+  const modules = orderBy(
+    packages.flatMap((p) => p.modules),
+    "name",
+  );
+
+  if (modules.length === 0) {
+    return <EmptyTabContent />;
+  }
+
+  return (
+    <Grid container spacing={2}>
+      <Grid
+        item
+        md={3}
+        sx={{
+          maxHeight: "500px",
+          overflowY: "auto",
+        }}
+      >
+        <ModuleSidebar
+          moduleNames={modules.map((m) => m.name)}
+          selectedModuleIndex={selectedModuleIndex}
+          setSelectedModuleIndex={setSelectedModuleIndex}
+        />
+      </Grid>
+      <Grid item md={9}>
+        <ModuleContent
+          address={address}
+          moduleName={modules[selectedModuleIndex].name}
+          sourceCode={modules[selectedModuleIndex].source}
+        />
+      </Grid>
+    </Grid>
+  );
+}
+
+function ModuleSidebar({
+  moduleNames,
+  selectedModuleIndex,
+  setSelectedModuleIndex,
+}: {
+  moduleNames: string[];
+  selectedModuleIndex: number;
+  setSelectedModuleIndex: (index: number) => void;
```

Comment:
lets pull it out as `ModuleSidebarProps` if it gets clumsy with couple variables?